### PR TITLE
Grass Planting C++ Solutions

### PR DESCRIPTION
**Pull request by sandywadhwa**
Added C++ solution - Intended Solution section
Ported DFS Solution from Java to C++ code

### CODE REVIEW

File: solutions/usaco-894.mdx [hunk 1]
```diff
@@ -20,6 +20,43 @@ $\max(\texttt{deg}[i])+1$ colors.
...
+		freopen((name+".out").c_str(), "w", stdout);
+	}
+}
+
+int main() {
+	setIO("planting");
+
+	int n; cin >> n;
+	int deg[n+1];
+	memset(deg, 0, sizeof(deg)); // Initialize all degrees to 0
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:44]: think you can do this w/ list initialization or smth
like `int deg[n + 1] {0}`
right @maggie-j-liu?

File: solutions/usaco-894.mdx [hunk 2]
```diff
@@ -20,6 +20,43 @@ $\max(\texttt{deg}[i])+1$ colors.
...
+	int u, v;
+	int maxDeg = INT_MIN;  // tracking maxDegree
+	for(int i=1;i<n;i++){
+		cin>>u>>v;
+		deg[u]++;
+		deg[v]++;
+		maxDeg = max(maxDeg, max(deg[u], deg[v]));
+	}
+	cout << (maxDeg+1) << "\n";
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:53]:
```suggestion
	cout << maxDeg + 1 << "\n";
```

File: solutions/usaco-894.mdx [hunk 3]
```diff
@@ -20,6 +20,43 @@ $\max(\texttt{deg}[i])+1$ colors.
...
+int main() {
+	setIO("planting");
+
+	int n; cin >> n;
+	int deg[n+1];
+	memset(deg, 0, sizeof(deg)); // Initialize all degrees to 0
+	int u, v;
+	int maxDeg = INT_MIN;  // tracking maxDegree
+	for(int i=1;i<n;i++){
+		cin>>u>>v;
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:48]: whitespace between operators please

File: solutions/usaco-894.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/usaco-894.mdx:44]: also whitespace between ops here

File: solutions/usaco-894.mdx [hunk 4]
```diff
@@ -90,6 +127,61 @@ valid assignment of grass types in the $\texttt{ans}$ array.
...
+	ios_base::sync_with_stdio(0); cin.tie(0);
+	if(name.size()){
+		freopen((name+".in").c_str(), "r", stdin);
+		freopen((name+".out").c_str(), "w", stdout);
+	}
+}
+
+bool visited[10001];
+int ans[10001];
+vector<int> adjList[10001];
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:148]: magic numbers are kinda bad
perhaps just have a `MAX_N`, set it to 10k, and have these be size `MAX_N + 1`

File: solutions/usaco-894.mdx [hunk 5]
```diff
@@ -90,6 +127,61 @@ valid assignment of grass types in the $\texttt{ans}$ array.
...
+
+void dfs(int self, int parent){
+	int len = adjList[self].size();
+	int g = 1;
+	for (int i : adjList[self]) {
+		if (i != parent) {
+			while (g == ans[self] || g == ans[parent]) g++;
+				ans[i] = g;
+				dfs(i, self);
+				g++;
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:158]: indentation here seems kinda weird

File: solutions/usaco-894.mdx [hunk 6]
```diff
@@ -90,6 +127,61 @@ valid assignment of grass types in the $\texttt{ans}$ array.
...
+		adjList[u].push_back(v);
+		adjList[v].push_back(u);
+	}
+	dfs(1, 1);
+	int maxAns = 0;
+	for (int curAns : ans) {
+		maxAns = max(maxAns, curAns);
+	}
+	cout << maxAns << "\n";
+	return 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:179]:
```suggestion
```

File: solutions/usaco-894.mdx [hunk 7]
```diff
@@ -90,6 +127,61 @@ valid assignment of grass types in the $\texttt{ans}$ array.
...
+				dfs(i, self);
+				g++;
+			}
+		}
+}
+
+int main() {
+	setIO("planting");
+
+	int n; cin >> n;
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:166]:
```suggestion
	int n;
	cin >> n;

File: solutions/usaco-894.mdx [hunk 8]
```diff
@@ -90,6 +127,61 @@ valid assignment of grass types in the $\texttt{ans}$ array.
...
+	int u, v;
+	for(int i=1;i<n;i++){
+		cin>>u>>v;
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:169]: whitespace between ops

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-894.mdx (hunk 1 above)
Comment (maggie-j-liu) [reply solutions/usaco-894.mdx:44]: `int deg[n + 1]{};` should work, but change the array to a constant size. variable length arrays are not c++ standard

File: solutions/usaco-894.mdx (hunk 6 above)
Comment (sandywadhwa) [reply solutions/usaco-894.mdx:179]: Little confused here, why not just return 0; when function return type is int.

Comment (maggie-j-liu) [reply solutions/usaco-894.mdx:179]: the main function returns 0 by default https://stackoverflow.com/questions/4138649/why-is-return-0-optional

File: solutions/usaco-894.mdx [hunk 9]
```diff
@@ -20,6 +20,45 @@ $\max(\texttt{deg}[i])+1$ colors.
...
 <LanguageSection>
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+#define MAX_N 10000
```
Comment (maggie-j-liu) [inline solutions/usaco-894.mdx:30]: would prefer `const int MAX_N = 10000;` https://usaco.guide/general/generic#macros

File: solutions/usaco-894.mdx [hunk 10]
```diff
@@ -90,6 +129,65 @@ valid assignment of grass types in the $\texttt{ans}$ array.
...
```
Comment (maggie-j-liu) [inline solutions/usaco-894.mdx:139]: same comment as above

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: solutions/usaco-894.mdx (hunk 10 above)
Comment (sandywadhwa) [reply solutions/usaco-894.mdx]: incorporated

File: solutions/usaco-894.mdx (hunk 9 above)
Comment (sandywadhwa) [reply solutions/usaco-894.mdx]: incorporated

**Review by maggie-j-liu — APPROVED** [review]

File: solutions/usaco-894.mdx [hunk 11]
```diff
@@ -90,6 +128,64 @@ valid assignment of grass types in the $\texttt{ans}$ array.
...
+	}
+}
+
+int main() {
+	setIO("planting");
+
+	int n; 
+	cin >> n;
+	int u, v;
+	for(int i = 1; i < n; i++){
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:173]: be consistent w/ spacing in control flow statements
```suggestion
	for (int i = 1; i < n; i++) {
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
just some more formatting stuff and we good

**Comment by sandywadhwa** [comment]
@SansPapyrus683 can you please review and let me know if there's anything else needed to get it through?

**Review by SansPapyrus683 — APPROVED** [review]
it's readable now ig

**Pull request merged**